### PR TITLE
feature/site-picker-refactor changes to trunk

### LINKED ISSUE #6111: Refactor SitePicker screens to MVVM
This master issue lists the changes we need to migrate the site picker screen to use MVVM. 
## FluxC
- [x] Migrate fetch supported woo version methods into a suspendable function ([#2331](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2331)).
- [x] Migrate fetch site settings methods into a suspendable function ([#2332](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2332)).
- [x] Migrate fetch site product settings methods into a suspendable function ([#2332](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2332)).

## Woo
- [x] Update existing methods in Woo that fetch site settings & product settings remove usages of [deprecated event bus architecture](https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/%5BDeprecated%5D-Architecture). (#6284 )
- [ ] Since the `SIGN OUT` action is the only action in the site picker that is utilising event bus, I thought of moving it to a separate class for now. It looks like the `AccountStore` is completely in Java and migrating the sign out action to suspendable is out of scope for this task.
- [x] Create a `SitePickerRepository` to communicate with FluxC to fetch site related details from the API.
- [x] Create a `ViewModel` that handles business logic for the Site picker screens.
- [x] Update the fragment classes and remove legacy code.

### PR DESCRIPTION
**Pull request by anitaa1990**
This PR closes #6111. It is the master branch includes changes from #6288 #6295 and #6298. All the changes have already been reviewed and all this PR does is merge those changes to trunk.

#### Switch Stores
- Click on the Hub menu icon -> Switch store button.
- Notice that the list of sites is displayed along with the user account info and the continue button.
- Click on one of the sites in the list and notice the progress dialog is displayed and the My Store tab is refreshed to reflect the currently selected site.

#### Login via site address: auto login to the correct site address
- Click on the `Login with site address` button. Enter a store url, email + password.
- Notice you are redirected to the site picker screen. 
- Progress dialog is displayed with `Verifying site` and you are redirected to the My Store tab for the entered site address.

#### Login via site address: auto login to a site address entered during login.
- Click on the `Login with site address` button. Enter a store url, email + password.
- Notice you are redirected to the site picker screen. 
- The current site is not found in the list of stores associated with the logged in user account so account mismatch view is displayed. 
- **If there are sites associated with the user account**, then `View connected stores` button is displayed.
- Click on `View Connected Stores` would redirect to a list of stores associated with the account.
- Click on `Try Another account` will redirect to the prologue screen after logging out the user.
- **If there are no sites associated with the user account**, then an empty view is displayed with the option to login with a different account and an option to install Jetpack on their store.
- **If the entered site address does not have WooCommerce installed**, then install Woo option is displayed to the merchant. They can install Woo and refresh the screen, if needed. 

#### Login via email to an account that does not have any stores:
- Click on the `Login with WordPress.com` button. Enter email + password.
- Notice you are redirected to the site picker screen. 
- Notice that if there are no sites associated with this account, then an empty view is displayed with the option to login with a different account and an option to install Jetpack on their store.
- Click on `Try Another account` will redirect to the prologue screen after logging out the user.

#### Jetpack CP flow
- The above testing steps should work for Jetpack CP sites too. It would be nice to test those scenarios using a Jetpack CP site.

#### Switch Store flow
. . 

#### Login via store address flow
. 

### Notes
There are no user facing changes as such. So I have not included any release notes to the PR. But it would be good to test these changes during the beta testing as well.

### CODE REVIEW

**Comment by anitaa1990** [comment]
@hichamboushaba I updated the `SitePickerView` to an enum class, per your suggestion 😊 I'm marking the PR as ready for review but I'll also be testing a lot to find any issues as such. But feel free to review if you can 🙏 Thank you!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt [hunk 1]
```diff
@@ -402,7 +402,8 @@ class SitePickerViewModel @Inject constructor(
         val isToolbarVisible: Boolean = false,
         val isProgressDiaLogVisible: Boolean = false,
         val isPrimaryBtnVisible: Boolean = false,
-        val isSecondaryBtnVisible: Boolean = false
+        val isSecondaryBtnVisible: Boolean = false,
+        val currentSitePickerEventView: SitePickerView = SitePickerView.StoreListView
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt]: np, unless I'm misunderstanding something, it seems the `event` part is a bit misleading, this is a state and not an event, and IMO it should be called as such. So something like: `sitePickerState` or simply `state`.

**Review by hichamboushaba — COMMENTED** [review]
Thanks @anitaa1990 for the the last changes, this looks really nice.

I didn't get a chance to test all cases, but I confirmed the issue I mentioned in last PR is fixed with last changes, I just left a np comment about the name of the variable, but feel free to ignore it if you don't agree with my thinking.

I'm not approving yet as I prefer to spend more time testing the other cases, and also it'd feel safer to merge this after code freeze, to give us a chance to catch any other bugs before releasing it to the beta group, WDYT?

**Comment by anitaa1990** [comment]
Thank you for testing and the review @hichamboushaba ! I'll work on the nitpick feedback now and totally ok with having to wait till Tuesday to merge to give us time to test those changes in `trunk` before release 👍

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt (hunk 1 above)
Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt]: Fixed in [23d6a6b](https://github.com/woocommerce/woocommerce-android/pull/6488/commits/23d6a6b7e0ac9fa6183f33fb11ba357c12b7a2b9)

File: build.gradle [hunk 2]
```diff
@@ -87,20 +87,20 @@ ext {
...
     assertjVersion = '3.11.1'
     aztecVersion = 'v1.3.45'
-    flipperVersion = '0.143.0'
+    flipperVersion = '0.138.0'
     stateMachineVersion = '0.2.0'
     coreKtxVersion = '1.7.0'
-    appCompatVersion = '1.4.1'
-    materialVersion = '1.5.0'
+    appCompatVersion = '1.4.0'
+    materialVersion = '1.4.0'
```
Comment (hichamboushaba) [inline build.gradle]: @anitaa1990 it seems the branch is bringing some unwanted changes, it downgrades some libraries that were updated in between.

Comment (anitaa1990) [reply build.gradle]: 🤦‍♀️ Thanks for catching that! Fixed in [46e9213](https://github.com/woocommerce/woocommerce-android/pull/6488/commits/46e921334ea3822c0588a9acca31952779bb23e9)

**Review by hichamboushaba — APPROVED** [review]
Looks good, nice work @anitaa1990.

As the code was reviewed in previous PRs, I didn't do a full code review here.

File: WooCommerce/src/main/res/navigation/nav_graph_main.xml [hunk 3]
```diff
@@ -240,8 +239,6 @@
     <include app:graph="@navigation/nav_graph_product_filters" />
     <include app:graph="@navigation/nav_graph_order_creations" />
     <include app:graph="@navigation/nav_graph_simple_payments" />
-    <include app:graph="@navigation/nav_graph_jetpack_install" />
-    <include app:graph="@navigation/nav_graph_coupons" />
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/navigation/nav_graph_main.xml:244]: @hafizrahman this is the cause of the crash, this was accidentally removed, I'll have a PR ASAP to reintroduce it.

**Pull request merged**